rules/no-assign-parameter.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * @fileoverview Do not assign to parameters.
+ * @author Simon Hajjar <simon.j.hajjar@gmail.com>
+ */
+
+'use strict';
+
+module.exports = {
+
+    meta: {
+
+        docs: {
+            recommended: true,
+            type: 'error',
+            description: 'Do not assign to parameters'
+        },
+
+        schema: []
+    },
+
+    create: function (context) {
+
+        function inspectFunctionDeclaration (emitted) {
+            if (emitted.exit) { return; }
+
+            var node = emitted.node;
+            let params = node.params.map(x => x['id']);
+            for (let statement of node.body.body) {
```

Comment:
in case of a fallback function, `node.body.body` will throw exception. Please account for this + add tests
first ensure that `node.is_abstract` is `false`